src/Rings/mpolyquo-localizations.jl

```diff
@@ -2565,7 +2565,41 @@ function small_generating_set(
   end::Vector{elem_type(base_ring(I))} 
 end
 
-dim(R::MPolyQuoLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyComplementOfPrimeIdeal}) = dim(saturated_ideal(modulus(R))) - dim(prime_ideal(inverted_set(R)))
+@attr Int function dim(R::MPolyQuoLocRing)
+  error("Not implemented")
+end
+
+@attr Int function dim(R::MPolyLocRing)
+  error("Not implemented")
+end
+
+@attr Int function dim(R::MPolyQuoLocRing{<:Any, <:Any, <:MPolyRing, <:MPolyRingElem, <:Union{MPolyComplementOfPrimeIdeal, MPolyComplementOfKPointIdeal}})
+  P = prime_ideal(inverted_set(R))
+  I = saturated_ideal(modulus(R))
+  return dim(I) - dim(P)
+end
```

Comment:
Please handle them separately. For MPolyComplementOfKPointIdeal P is certified to be zero-dimensional so that we do not need to compute dim(P).

Reply:
> P is certified to be zero-dimensional

It seems that this is not true.

```
julia> R, (x, y) = ZZ[:x, :y]
(Multivariate polynomial ring in 2 variables over ZZ, ZZMPolyRingElem[x, y])

julia> S = complement_of_point_ideal(R, [0, 0])
Complement
  of maximal ideal corresponding to rational point with coordinates (0, 0)
  in multivariate polynomial ring in 2 variables over ZZ

julia> I = ideal(R, 0)
Ideal generated by
  0

julia> L, _ = localization(R, S)
(Localization of multivariate polynomial ring in 2 variables over ZZ at complement of maximal ideal of point (0, 0), Hom: R -> localized ring)

julia> W, _ = quo(L, L(I))
(Localization of quotient of multivariate polynomial ring at complement of maximal ideal, hom: Localized ring -> Localized quotient of multivariate polynomial ring)

julia> P = prime_ideal(inverted_set(W))
Ideal generated by
  x
  y

julia> dim(P)
1
```

So I will leave it as it is for now.

Reply:
My guess is that `complement_of_point_ideal` silently assumes that we are working over a field? Perhaps the type and constructor should be restricted accordingly what do you think @afkafkafk13 ?

Reply:
> julia> S = complement_of_point_ideal(R, [0, 0])
> Complement
>   of maximal ideal corresponding to rational point with coordinates (0, 0)
>   in multivariate polynomial ring in 2 variables over ZZ

This is indeed a problem with complement_of_point_ideal: 
This ideal is NOT maximal as we do not get a field as the quotient of the ring by it. This needs to be handled in `complement_of_point_ideal`. My suggestion: check dimension (not number of entries!!) of the ideal before doing anything else in `complement_of_point_ideal`, throw an error if it is not maximal and point the user to `complement_of_prime_ideal`.

Reply:
It is a bit fuzzy what a "point" is here. Will we get anything computationally effective for the "point" given by the maximal ideal `(2,x,y)` in `Z[x,y]`? I impression was that `ComplementOfKPointIdeal` just assumes a base field.

Reply:
> I impression was that ComplementOfKPointIdeal just assumes a base field.

I think it does assume but does not check for this (although it should).

I would leave fixing `complement_of_point_ideal` to a separate issue.

Reply:
ok. fine with me. But we urgently need to fix complement_of_point_ideal. Maybe we can just check

@simonbrandhorst no, we won't get anything computationally effective currently. You are right. We should just tell the user to use `complement_of_prime_ideal` instead, whenever we are not over a field.